Jenkinsfile

```diff
@@ -132,19 +132,61 @@ pipeline {
                         sh 'poetry run sphinx-build -W -b html docs docs/_build/html'
                     }
 
+                    dir('external-docs'){
+                        sh "python3 -m mkdocs build"
+                    }
+
                     stash name: 'docs', includes: '**/docs/_build/html/**/*'
+                    stash name: 'mkdocs', includes: '**/external-docs/site/**/*'
                 }
             }
         }
-        stage('Build Static Site'){    
+        stage('Publishing Documentation'){
+            when{
+                branch 'main'
+            }
             steps{
                 script{
-                    dir('external-docs'){
-                        sh "python3 -m mkdocs build"
+                    try {
+                        withCredentials([gitUsernamePassword(credentialsId: 'csvcubed-github', gitToolName: 'git-tool')]){
+                            sh 'git clone "https://github.com/GSS-Cogs/csvcubed-docs.git"'
+                            dir ('csvcubed-docs') {
+                                sh 'git config --global user.email "csvcubed@gsscogs.uk" && git config --global user.name "csvcubed"'
+                                
+                                if (fileExists("external")) {
+                                    sh 'git rm -rf external'
+                                }
+                                sh 'mkdir external'
+                                sh 'cp -r ../external-docs/site/* external'
+
+                                if (fileExists("api-docs")) {
+                                    sh 'git rm -rf api-docs'
+                                }
+                                sh 'mkdir api-docs'
+                                sh 'mkdir api-docs/csvcubed'
+                                sh 'mkdir api-docs/csvcubed-devtools'
+                                sh 'mkdir api-docs/csvcubed-models'
+                                sh 'mkdir api-docs/csvcubed-pmd'
+
+                                sh 'cp -r ../csvcubed/docs/_build/html/* api-docs/csvcubed'
+                                sh 'cp -r ../csvcubed-devtools/docs/_build/html/* api-docs/csvcubed-devtools'
+                                sh 'cp -r ../csvcubed-models/docs/_build/html/* api-docs/csvcubed-models'
+                                sh 'cp -r ../csvcubed-pmd/docs/_build/html/* api-docs/csvcubed-pmd'
+
+                                sh 'touch .nojekyll'
+
+                                sh 'git add *'
+                                sh 'git add .nojekyll'
+                                sh 'git commit -m "Updating documentation."'
+                                sh 'git checkout gh-pages'
```

Comment:
Did the publish work without using the `gh-pages` branch?

Reply:
yes. It ends up that without these lines, no GitHub workflows get activated, so the csvcubed-docs repo doesn't build and deploy the latest files pushed to it unless you have these lines of code present.